plugins/org.eclipse.fordiac.ide.systemconfiguration/src/org/eclipse/fordiac/ide/systemconfiguration/editor/SystemConfigurationEditor.java

```diff
@@ -55,12 +55,24 @@ public SystemConfiguration getModel() {
 
 	@Override
 	public void setInput(final IEditorInput input) {
-		if (input instanceof final SystemConfigurationEditorInput sysConfInput) {
-			sysConf = sysConfInput.getContent();
-		}
+		checkEditorInput(input);
+		final SystemConfigurationEditorInput sysConfInput = (SystemConfigurationEditorInput) input;
```

Comment:
I would suggest returning the `SystemConfigurationEditorInput` from `checkEditorInput(...)` to avoid the cast.

Reply:
good point, I had something similar in mind for changes proposed by  @oberlehner below.